libvips/colour/icc_transform.c

```diff
@@ -840,44 +839,50 @@ vips_icc_import_build( VipsObject *object )
 	return( 0 );
 }
 
-static void 
+static void
 decode_lab( guint16 *fixed, float *lab, int n )
 {
 	int i;
 
-        for( i = 0; i < n; i++ ) {
-                lab[0] = (double) fixed[0] / 652.800;
-                lab[1] = ((double) fixed[1] / 256.0) - 128.0;
-                lab[2] = ((double) fixed[2] / 256.0) - 128.0;
+	for( i = 0; i < n; i++ ) {
+		cmsCIELab Lab;
+		cmsLabEncoded2Float( &Lab, fixed );
```

Comment:
Can a compiler optimise things here, or will this always result in a shared library call per-pixel?

(The lcms source appears to use slightly different constants for this calculation e.g. https://github.com/mm2/Little-CMS/blob/9ca6b591e0af9bebe6412dba907cb077f6291a8c/src/cmspcs.c#L204-L215 )

Reply:
I inlined these calculations with commit 91070f0810f453995f6fcb3f91474600ea9c883d, allowing compilers to optimise.

lcms has two APIs to decode a Lab value:
- `cmsLabEncoded2Float()` - decode a Lab value, encoded on ICC v4 convention.
- `cmsLabEncoded2FloatV2()` - decode a Lab value, encoded on ICC v2 convention.

The reason I used the former as the reference is to match the `cmsCreateLab4Profile()` API.  Previously, libvips used the former as the reference, which is probably(?) only appropriate when using `cmsCreateLab2Profile()`.